javascripts/discourse/components/dc-categories-nav.js.es6

```diff
@@ -15,12 +15,27 @@ export default Component.extend({
 
   customLinks: computed(function() {
     const links = settings.header_categories_custom_links.split("|");
-    return links.map(entry => {
-      const [text, url] = entry.split(",").map(str => str.trim());
+
+    const linkItems = links.map(entry => {
```

Comment:
Everything looks great @duranmla! I just have a couple of suggestions on the code style

What do you think about the following?

```js
computed(function() {
  const links = settings
    .header_categories_custom_links
    .split("|");

  function isEntryValid(entry) {
    return entry.trim();
  }

  function isParsedEntryValid(parsedEntry) {
    return parsedEntry.length === 2;
  }

  const linkItems = links.map(entry => {
    if (!isEntryValid(entry)) return null;

    const parsedEntry = entry.split(",");

    if (!isParsedEntryValid(parsedEntry)) return null;

    const [text, url] = parsedEntry.map(str => str.trim());

    if (!text || !url) return null;

    const target = this._getAnchorTag(url);

    return { text, url, target };
  });

  return linkItems.filter(Boolean);
});
```

Totally up to you

Reply:
I do see your point and it is very valuable but in this case I will pass. **Nevertheless,** what we/I should/will do is to implement a way to iterate properly over the code _(testing)_. I think I made a mistake at the beginning of this development to not try harder with a way to test at least unit test for the functions we create to customise Discourse _(discourse has a lot of limitations)_.

Often the changes were aesthetics matters and there wasn't much logic. My point is that this refactors are currently _"hard"_ because there is no indications if you break something or not, so devs have to re-test everything before submit _( but without your help I wouldn't wake up and realise that we are failing to refactor properly due to the lack of trust)_

Thanks a lot!